src/components/Dashboard/RewardsModule.js

```diff
@@ -88,28 +116,42 @@ const RewardsModule = React.memo(function RewardsModule({
         }
 
         // Withdraw funds from treasury
-        rewardTransactionQueue.push(
-          await onWithdraw(feeToken.id, wallet.account, totalFees)
-        )
+        if (totalTreasuryFees.gt(0)) {
+          rewardTransactionQueue.push(
+            await onWithdraw(feeToken.id, wallet.account, totalTreasuryFees)
+          )
+        }
+
+        // Claim subscription fees
+        for (const subscriptionFee of subscriptionFees) {
+          rewardTransactionQueue.push(
+            await onClaimSubscriptionFees(subscriptionFee.periodId)
+          )
+        }
 
         await Promise.all(rewardTransactionQueue.map(tx => tx.wait()))
+
+        setSubscriptionFees([])
       } catch (err) {
         console.error(`Error claiming rewards: ${err}`)
+        Sentry.captureException(err)
```

Comment:
Just wondering; do we need to guard this with a check that Sentry is actually initialized?

Reply:
I don't think we need to, just tried as well capturing an error without initializing, with no issues